### DISCUSSION TITLE
前端报错 网络错误(TypeError) / JSON解析错误(SyntaxError)

### DISCUSSION THREAD

**Discussion by anghunk** [ArtalkJS/Artalk#82] [Q&A] [answered]
博客程序：nuxt + vue 

**artalk后端已经部署完成**

![image](https://user-images.githubusercontent.com/65840178/154222579-fb542ab9-b5e8-4a5e-ae88-b25248858c57.png)
![image](https://user-images.githubusercontent.com/65840178/154222625-1de16cee-75ee-4070-9b85-7b1e1cdba12f.png)
![image](https://user-images.githubusercontent.com/65840178/154223065-cf187a3e-8caf-4106-b486-e67f292c5252.png)

**前端部署**

```vue
<template>
  <div class="wrapper">
      <link href="https://cdn.jsdelivr.net/npm/artalk@2.1.4/dist/Artalk.css" rel="stylesheet" />
      <script src="https://cdn.jsdelivr.net/npm/artalk@2.1.4/dist/Artalk.js"></script>
      <div id="Comments"></div>
    </div>
</template>

<script>
const el = '#Comments';
const pageKey = '';
const pageTitle = '';
const server = 'https://artalk.imhan.cn/api/';
const site = '博客';
export default {
  name: 'Comments',
  mounted () {
    this.comments()
  },
  methods: {
    comments () {
      try {
        new Artalk({
          el,
          pageKey,
          pageTitle,
          server,
          site,
        })
      } catch (e) { }
    },
  }
}
</script>

<style lang="scss" scoped>
// ...
</style>
```

**控制台报错**

![image](https://user-images.githubusercontent.com/65840178/154223029-9c187156-08da-4fda-bd00-653adf8baeec.png)

```js
Access to fetch at 'https://artalk.imhan.cn/api/get' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
Artalk.js:1          POST https://artalk.imhan.cn/api/get net::ERR_FAILED 404

Artalk.js:1 Uncaught (in promise) Error: 网络错误
    at Artalk.js:1:51547
    at Generator.throw (<anonymous>)
    at a (Artalk.js:1:819)

```

**Comment by anghunk** [comment]
原来前端的php版本指向的是`index.php` ，我直接看文档看的`/api`是go版本

然后出现了这个问题，记录一下，正在研究中

![image](https://user-images.githubusercontent.com/65840178/154412698-d64c3a03-4340-45c7-8f39-3a757ed94ede.png)

**Comment by qwqcode** [comment, marked as answer]
看了你的代码，前端引入的是 artalk@2.1.4，而后端 https://artalk.imhan.cn 是 PHP 旧版
前端目前已经不再支持 Artalk 的 PHP 后端，你需要部署 Golang 新版
可参考：
https://artalk.js.org/guide/backend/install.html
https://www.bilibili.com/video/BV1cP4y1J7gC
